src/mesh/boundary_standard.cxx

```diff
@@ -23,473 +21,168 @@
-
-    //Now decide if it's a local or global limit, prefer global if a tie
-    if(ptsAvailGlobal <= ptsAvailLocal){
-      ptsAvail = ptsAvailGlobal;
-      gridType = "global";
-    }else{
-      ptsAvail = ptsAvailLocal;
-      gridType = "local";
+    Mesh* localmesh = region->localmesh;
+
+    int ptsAvailGlobal, ptsAvailLocal, ptsAvail;
+    string side, gridType;
+
+    //Initialise var in case of no match and CHECK<=2
+    ptsAvail = ptsRequired; //Ensures test passes without exception
+
+    switch(region->location) {
+    case BNDRY_XIN:
+    case BNDRY_XOUT: {
+      side = "x";
+
+      //Here 2*localmesh->xstart is the total number of guard/boundary cells
+      ptsAvailGlobal = localmesh->GlobalNx - 2*localmesh->xstart;
+
+      //Work out how many processor local points we have excluding boundaries
+      //but including ghost/guard cells
+      ptsAvailLocal  = localmesh->LocalNx;
+      if(localmesh->firstX()) ptsAvailLocal -= localmesh->xstart;
+      if(localmesh->lastX())  ptsAvailLocal -= localmesh->xstart;
+
+      //Now decide if it's a local or global limit, prefer global if a tie
+      if(ptsAvailGlobal <= ptsAvailLocal){
+        ptsAvail = ptsAvailGlobal;
+        gridType = "global";
+      }else{
+        ptsAvail = ptsAvailLocal;
+        gridType = "local";
+      }
+
+      break;
     }
+    case BNDRY_YUP:
+    case BNDRY_YDOWN: {
+      side = "y";
 
-    break;
-  }
-  case BNDRY_YUP: 
-  case BNDRY_YDOWN: {
-    side = "y";
-
-    //Here 2*mesh->ystart is the total number of guard/boundary cells
-    ptsAvailGlobal = mesh->GlobalNy - 2*mesh->ystart;
-
-    //Work out how many processor local points we have excluding boundaries
-    //but including ghost/guard cells
-    ptsAvailLocal  = mesh->LocalNy;
-    if(mesh->firstY()) ptsAvailLocal -= mesh->ystart;
-    if(mesh->lastY())  ptsAvailLocal -= mesh->ystart;
-
-    //Now decide if it's a local or global limit, prefer global if a tie
-    if(ptsAvailGlobal <= ptsAvailLocal){
-      ptsAvail = ptsAvailGlobal;
-      gridType = "global";
-    }else{
-      ptsAvail = ptsAvailLocal;
-      gridType = "local";
+      //Here 2*localmesh->ystart is the total number of guard/boundary cells
+      ptsAvailGlobal = localmesh->GlobalNy - 2*localmesh->ystart;
+
+      //Work out how many processor local points we have excluding boundaries
+      //but including ghost/guard cells
+      ptsAvailLocal  = localmesh->LocalNy;
+      if(localmesh->firstY()) ptsAvailLocal -= localmesh->ystart;
+      if(localmesh->lastY())  ptsAvailLocal -= localmesh->ystart;
+
+      //Now decide if it's a local or global limit, prefer global if a tie
+      if(ptsAvailGlobal <= ptsAvailLocal){
+        ptsAvail = ptsAvailGlobal;
+        gridType = "global";
+      }else{
+        ptsAvail = ptsAvailLocal;
+        gridType = "local";
+      }
+
+      break;
+    }
+#if CHECK > 2 //Only fail on Unrecognised boundary for extreme checking
```

Comment:
I know this isn't part of this PR, but I forget the rationale for turning this on only at high checking. The check is free, so what situation would we not want to check this?

Reply:
I'll take out this `#if CHECK > 2` for now. It's easy to remove the commit if anyone remembers why it was there.